mart/attack/adversary.py

```diff
@@ -132,8 +132,8 @@ def done(self) -> bool:
 
         return False
 
-    def on_run_start(self, adversary, input, target, model, **kwargs):
-        super().on_run_start(adversary, input, target, model, **kwargs)
+    def on_run_start(self, adversary, **kwargs):
```

Comment:
I think you should be explicit with the parameters here. That is, explicit define what parameters you need rather than hiding them in `kwargs`. This will catch more bugs.